.github/workflows/nx-cloud-agents.yml

```diff
@@ -48,44 +51,63 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - name: Detect package manager
+        id: package_manager
+        run: |
+          echo "::set-output name=name::$([[ -f ./yarn.lock ]] && echo "yarn" || ([[ -f ./pnpm-lock.yaml ]] && echo "pnpm") || ([[ -f ./package-lock.json ]] && echo "npm") || echo "N/A")"
```

Comment:
I can't remember the last time I saw a project without a lock file and there really isn't any reason to attempt to support it, especially as this workflow is specific to nx, and nx will generate one for you, so folks we have to be actively working to have a worse setup. I think the default should simply be `npm` here and we should run `npm ci`. If that then errors, it is an appropriate error